### PR TITLE
Added support for all resolution types in GDAX downloader.

### PR DESCRIPTION
**Pull request by manisin**
Supported the daily/hour/minute/second resolution for gdax downloader.
Optimized data downloaded per request to 200 datapoints (the max supported by GDAX)

### CODE REVIEW

File: ToolBox/GDAXDownloader/GDAXDownloader.cs [hunk 1]
```diff
@@ -12,26 +12,32 @@
...
-using QuantConnect.Data.Market;
-using Newtonsoft.Json;
-using System.Linq;
-using System.Text;
-using Newtonsoft.Json.Linq;
-using QuantConnect.Logging;
 
 namespace QuantConnect.ToolBox.GDAXDownloader
 {
+    using System;
```
Comment (jaredbroad) [inline ToolBox/GDAXDownloader/GDAXDownloader.cs]: Normally we've kept the using statements outside the namespace. Shouldn't be needed here.

File: ToolBox/GDAXDownloader/GDAXDownloader.cs [hunk 2]
```diff
@@ -41,77 +47,101 @@ public class GDAXDownloader : IDataDownloader
...
+                        Value = decimal.Parse(datapoint[4].ToString()),
+                        DataType = MarketDataType.TradeBar,
+                        Period = new TimeSpan(0, 0, (int)granularity),
+                        EndTime = Time.UnixTimeStampToDateTime(epochs).AddSeconds(granularity)
+                    };
+                    returnData.Add(tradeBar);
                 }
             }
-            return returnData;
+            return returnData.OrderBy(datapoint => datapoint.Time).ToList();
```
Comment (jaredbroad) [inline ToolBox/GDAXDownloader/GDAXDownloader.cs]: Curious did you ever see this coming back in the wrong order?

File: ToolBox/GDAXDownloader/GDAXDownloader.cs [hunk 3]
```diff
@@ -41,77 +47,101 @@ public class GDAXDownloader : IDataDownloader
...
-            }
+            DateTime windowStartTime = startUtc;
+            DateTime windowEndTime = startUtc.AddSeconds((MaxDatapointsPerRequest - 1) * granularity);
+            windowEndTime = windowEndTime > endUtc ? endUtc : windowEndTime;
 
-            var counter = startUtc;
-            const string url = "http://api.gdax.com/products/{0}/candles?start={1}&end={2}&granularity={3}";
-
-            while (counter <= endUtc)
+            while (windowStartTime != windowEndTime)
```
[the remark this answers is not in the document]
Comment (manisin) [reply ToolBox/GDAXDownloader/GDAXDownloader.cs]: Done!

File: ToolBox/GDAXDownloader/GDAXDownloader.cs (hunk 2 above)
Comment (manisin) [reply ToolBox/GDAXDownloader/GDAXDownloader.cs]: The gdax response returns the data from latest->oldest order, thereby requiring reordering before piping it to the writer.

**Pull request merged**